### PR TITLE
underbar should respect safe area by default, with option to override

### PR DESCRIPTION
**Pull request by mcleinman**
**Fixes Phabricator ticket:** part of already-completed https://phabricator.wikimedia.org/T261979

### Notes
* This is being pushed to 6.7.1 to have some more testing time.

### Test Steps
1. Run app on a phone/simulator with a notch.
1. Open On This Day VC (from Explore feed or deep link).
1. Rotate phone so notch is on leading edge. Ensure the On This Day header respects the safe area. 
1. Open a reading list (on saved tab). Rotate phone so notch is on leading edge. Ensure the title header (`x articles`/`[title of list]`) respects the safe area.
1. Edit a section of any article, tap the insert image icon.
1. Rotate phone so notch is on leading edge. Ensure the header goes edge-to-edge (does NOT respect the safe area).

### CODE REVIEW

**Comment by mcleinman** [comment]
On hold, as this is destined for 6.7.1

**Comment by mcleinman** [comment]
Off hold

File: WMF Framework/NavigationBar.swift [hunk 1]
```diff
@@ -361,6 +361,20 @@ public class NavigationBar: SetupView, FakeProgressReceiving, FakeProgressDelega
...
+        didSet {
+            guard shouldUnderBarIgnoreSafeArea != oldValue else {
+                return
+            }
+
+            safeAreaLayoutGuide.leadingAnchor.constraint(equalTo: underBarView.leadingAnchor).isActive = !shouldUnderBarIgnoreSafeArea
+            safeAreaLayoutGuide.trailingAnchor.constraint(equalTo: underBarView.trailingAnchor).isActive = !shouldUnderBarIgnoreSafeArea
+
+            leadingAnchor.constraint(equalTo: underBarView.leadingAnchor).isActive = shouldUnderBarIgnoreSafeArea
+            trailingAnchor.constraint(equalTo: underBarView.trailingAnchor).isActive = shouldUnderBarIgnoreSafeArea
```
Comment (joewalsh) [inline WMF Framework/NavigationBar.swift:375]: `constraint(equalTo:` creates a new constraint and doesn't return an existing one, so new constraints will be created each time `shouldUnderBarIgnoreSafeArea` is set. Could these be created and stored in properties in `setup()`?
```
underBarRespectSafeAreaConstraints = [underBarViewLeadingConstraint, underBarViewTrailingConstraint]
underBarIgnoreSafeAreaConstraints = [...]
```
Then you could toggle `isActive` on those arrays here

**Review by joewalsh — CHANGES_REQUESTED** [review]

File: WMF Framework/NavigationBar.swift (hunk 1 above)
Comment (mcleinman) [reply WMF Framework/NavigationBar.swift:375]: My bad - I had it in my head that identically created constraints were treated as if they were the same. Not sure where I got that from. Will update this shortly.

**Review by joewalsh — APPROVED** [review]

**Pull request merged**